### PR TITLE
refactor wasm code to split reusable stuff from per-thread stuff

### PR DESCRIPTION
**Pull request by ajwootto**

### CODE REVIEW

**Comment by ajwootto** [comment]
Current dependencies on/for this PR:
* main
  * **PR #103** <a href="https://app.graphite.dev/github/pr/DevCycleHQ/go-server-sdk/103" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/DevCycleHQ/go-server-sdk/103?utm_source=stack-comment).

File: localbucketing.go [hunk 1]
```diff
@@ -62,38 +61,30 @@ type DevCycleLocalBucketing struct {
...
-	d.wasmStore = wasmtime.NewStore(wasmtime.NewEngine())
-	d.wasmStore.SetWasi(d.wasiConfig)
-	d.wasmLinker = wasmtime.NewLinker(d.wasmStore.Engine)
-	err = d.wasmLinker.DefineWasi()
 
-	if err != nil {
-		return
-	}
+	d.wasmStore = wasmtime.NewStore(d.wasmMain.wasmEngine)
+	d.wasmStore.SetWasi(d.wasiConfig)
```
Comment (JamieSinn) [inline localbucketing.go:76]: Due to the recent asc upgrade we probably can get rid of the WASI config - might be better suited for a different PR though.

**Review by JamieSinn — APPROVED** [review]
